src/config/internal.cpp

```diff
@@ -19,6 +48,26 @@ void check_session_id(std::string_view session_id, std::string_view prefix) {
                 "; got " + std::string{session_id}};
 }
 
+SessionIdPrefix get_session_id_prefix(std::string_view id) {
+    std::string prefix_list_str = "";
+
+    if (oxenc::is_hex(id)) {
+        for (auto prefix : all_session_id_prefixes) {
+            auto prefix_str = to_string(prefix);
+            prefix_list_str += (prefix_list_str.empty() ? prefix_str : ", {}"_format(prefix_str));
+
+            if ((id.size() == 64 + prefix_str.size() &&
+                 id.substr(0, prefix_str.size()) == prefix_str))
+                return prefix;
+        }
+    }
```

Comment:
```suggestion
    if (oxenc::is_hex(id) && id.size() == 66)
        for (auto prefix : all_session_id_prefixes) {
            if (id.starts_with(to_string(prefix)))
                return prefix;
```

Reply:
The vast majority of the time don't need `prefix_list_str`, so seems better to make it only if you run off the loop by ignoring it here, and then (see next suggestion).